index.js

```diff
@@ -12,8 +12,46 @@ var unlink = RSVP.denodeify(fs.unlink);
 var tmpDir = require('os').tmpdir();
 var debug = require('debug')('async-disk-cache');
 var zlib = require('zlib');
+var heimdall = require('heimdalljs');
 
 var CacheEntry = require('./lib/cache-entry');
+var Metric = require('./lib/metric');
+
+if (!heimdall.hasMonitor('async-disk-cache')) {
+  heimdall.registerMonitor('async-disk-cache', function AsyncDiskCacheSchema() {});
+}
+
+/*
+ * @private
+ *
+ * Defines a function on the given object at the given property name. Wraps
+ * the function with metric recording for heimdalljs.
+ *
+ * @method defineFunction
+ * @param Object obj the object on which to define the function
+ * @param String name the name to use for the function
+ * @param Function fn
+ * @returns Void
+ */
+function defineFunction(obj, name, fn) {
+  obj[name] = function() {
+    var stats = heimdall.statsFor('async-disk-cache');
+    var metrics = stats[name] = stats[name] || new Metric();
```

Comment:
if the monitor is responsible for this, we won't need to do this check every time.